tests/integration/gen_diff_test.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import libnmstate
+
+from .testlib.yaml import load_yaml
+
+
+def test_gen_diff():
+    des_state = load_yaml(
+        """---
+        interfaces:
```

Comment:
If `gen_diff` is also applicable to DNS settings (only generate the dns-resolver difference between the current state and desired state)? If not, `pytest --dump-states-with-description-only` will include all the DNS settings in the YAML files.

Reply:
Yes. The `pytest --dump-states-with-description-only` will only dump diff YAML.